src/gui/properties/peerlistwidget.cpp

```diff
@@ -188,34 +188,28 @@ void PeerListWidget::displayToggleColumnsMenu(const QPoint &)
         if ((i == PeerListColumns::COUNTRY) && !Preferences::instance()->resolvePeerCountries())
             continue;
 
-        QAction *myAct = menu->addAction(m_listModel->headerData(i, Qt::Horizontal, Qt::DisplayRole).toString());
-        myAct->setCheckable(true);
-        myAct->setChecked(!isColumnHidden(i));
-        myAct->setData(i);
-    }
-
-    connect(menu, &QMenu::triggered, this, [this](const QAction *action)
-    {
-        int visibleCols = 0;
-        for (int i = 0; i < PeerListColumns::IP_HIDDEN; ++i)
+        const auto columnName = m_listModel->headerData(i, Qt::Horizontal, Qt::DisplayRole).toString();
+        QAction *action = menu->addAction(columnName, this, [this, i](const bool checked)
         {
-            if (!isColumnHidden(i))
-                ++visibleCols;
-
-            if (visibleCols > 1)
-                break;
-        }
+            if (!checked && (visibleColumnsCount() <= 1))
+                return;
 
-        const int col = action->data().toInt();
+            setColumnHidden(i, !checked);
 
-        if (!isColumnHidden(col) && (visibleCols == 1))
-            return;
+            if (checked && (columnWidth(i) <= 5))
+                resizeColumnToContents(i);
 
-        setColumnHidden(col, !isColumnHidden(col));
-
-        if (!isColumnHidden(col) && (columnWidth(col) <= 5))
-            resizeColumnToContents(col);
+            saveSettings();
+        });
+        action->setCheckable(true);
+        action->setChecked(!isColumnHidden(i));
+    }
 
+    menu->addSeparator();
+    menu->addAction(tr("Resize all columns"), this, [this]()
```

Comment:
>Maybe "Adjust all columns size"? "Resize" is too generic, IMO (just like "change size").

It doesn't sound more specific/explanatory than the current one. So since they both have the same meaning to me, i prefer the current one.

Reply:
"Adjust all column widths"?

Reply:
>It doesn't sound more specific/explanatory than the current one.

Same here.

I was trying to make it shorter so it would look better with other options on menu, but it seems there are different opinions. So what about using the full description: "Resize column width to contents"?

Reply:
> I was trying to make it shorter so it would look better with other options on menu

👍 
>So what about using the full description: "Resize column width to contents"?

Can you add tooltip instead?

Reply:
>Can you add tooltip instead?

Done.